### ISSUE TITLE
Using data augmentation wrecks validation accuracy.

### ISSUE THREAD

**Issue by adamkc** [rstudio/keras#432]
I initially found this issue with my own models but it is repeatable using the cifar10_cnn example (from: keras/vignettes/examples/cifar10_cnn.R)

Run the code for the above example as is and you'll find the validation accuracy doesn't increase (despite training accuracy increasing) and validation loss is very high.  Then set line 7 `data_augmentation = FALSE` and you'll find validation accuracy and loss are working as expected. 

If I comment out all the `image_data_generator` arguments except `horizontal_flip` the script works as expected so the issue is produced only by some of the `image_data_generator` arguments.

----------------------------------------------
My session Info:

> ```
> > sessionInfo()
> R version 3.4.2 (2017-09-28)
> Platform: x86_64-w64-mingw32/x64 (64-bit)
> Running under: Windows >= 8 x64 (build 9200)
> 
> Matrix products: default
> 
> locale:
> [1] LC_COLLATE=English_United States.1252  LC_CTYPE=English_United States.1252    LC_MONETARY=English_United States.1252
> [4] LC_NUMERIC=C                           LC_TIME=English_United States.1252    
> 
> attached base packages:
> [1] stats     graphics  grDevices utils     datasets  methods   base     
> 
> other attached packages:
> [1] keras_2.1.6
> 
> loaded via a namespace (and not attached):
>  [1] compiler_3.4.2   magrittr_1.5     R6_2.2.2         Matrix_1.2-11    tools_3.4.2      whisker_0.3-2    base64enc_0.1-3  yaml_2.1.19     
>  [9] Rcpp_0.12.16     reticulate_1.7.1 tensorflow_1.5   grid_3.4.2       zeallot_0.1.0    jsonlite_1.5     tfruns_1.3       lattice_0.20-35 
> ```

**Comment by adamkc** [comment]
Can someone else run the keras/vignettes/examples/cifar10_cnn.R example to help me determine if the data augmentation issues are local to my machine.  Or if that is obvious from my description above does anyone have any recommendations on how to trouble shoot this problem?  Reinstall Keras?

**Comment by skeydan** [comment]
Hi,

I think you might have found something here :-)

Can you please comment just these 2

 featurewise_center = TRUE,
 featurewise_std_normalization = TRUE,

and confirm it works fine now?

Thanks!

**Comment by adamkc** [comment]
Hi @skeydan , After commenting out just the two that you suggest above the model does seem to train as expected.  Both accuracy and validation_accuracy increase with each epoch.  Do you have a guess on what this might mean?

**Comment by skeydan** [comment]
Yeah, these two parameters standardize the pixel values (pixel values divided by 255 that is, so values between 0 and 1, which now get shifted to a mean of 0).
Now the test set doesn't get  standardized and so, the pictures look fundamentally different to the network.

We'll update the example accordingly, thanks again for pointing this out!
